src/main/java/com/jcabi/aether/RepositoryAuthentication.java

```diff
@@ -67,11 +67,16 @@
      * Creates a new authentication with the specified properties.
      * @param auth The authentication object.
      */
+    @SuppressWarnings("PMD.ConfusingTernary")
```

Comment:
the if/else approach:

    if (auth.getPassword() != null) {
      this.password = auth.getPassword().toCharArray();
    } else {
      this.password = null;
    }
    if (auth.getPassphrase() != null) {
      this.passphrase = auth.getPassphrase().toCharArray();
    } else {
      this.passphrase = null;
    }

will fail these rules:

    [ERROR] /src/main/java/com/jcabi/aether/RepositoryAuthentication.java[72-76]: Avoid if (x != y) ..; else ..; (ConfusingTernary)
    [ERROR] /src/main/java/com/jcabi/aether/RepositoryAuthentication.java[78-82]: Avoid if (x != y) ..; else ..; (ConfusingTernary)
    [ERROR] /src/main/java/com/jcabi/aether/RepositoryAuthentication.java[75-75]: Assigning an Object to null is a code smell.  Consider refactoring. (NullAssignment)
    [ERROR] /src/main/java/com/jcabi/aether/RepositoryAuthentication.java[81-81]: Assigning an Object to null is a code smell.  Consider refactoring. (NullAssignment)

@pinaf I think the tenary operator approach is more elegant and shorter (2 lines vs 5).